Docs Audit #2 (#3801)
dfinity/portal: docs/developer-docs/smart-contracts/maintain/delete.mdx

import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
import { GlossaryTooltip } from "/src/components/Tooltip/GlossaryTooltip";

# Delete

<MarkdownChipRow labels={["Beginner", "Concept"]} />

## Overview

> If you want to permanently delete a specific <GlossaryTooltip>canister</GlossaryTooltip> or all canisters for a project, you can use the [`dfx canister delete`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-delete) command.

Author: ```suggestion
If you want to permanently delete a specific <GlossaryTooltip>canister</GlossaryTooltip> or all canisters for a project, you can use the [`dfx canister delete <canister-name>`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-delete) command.
```

> Deleting a canister removes the canister identifier, code, and state. Before you can delete a canister, you must first stop the canister to clear any pending message requests or replies.

Author: ```suggestion
Deleting a canister removes the canister identifier, code, and state. Before you can delete a canister, you must first stop the canister using [`dfx canister stop <canister-name>`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-stop) to clear any pending message requests or replies.
```

> Deleting a canister removes the canister identifier, code, and state. Before you can delete a canister, you must first stop the canister using [`dfx canister stop <canister-name>`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-stop) to clear any pending message requests or replies.

Reviewer: ```suggestion
Deleting a canister permanently removes its code and state. This action cannot be undone and once the canister is deleted, its ID cannot be reused. Before you can delete a canister, you must first stop the canister using [`dfx canister stop <canister-name>`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-stop) to clear any pending message requests or replies.
```

Reviewer: @jessiemongeon1 I am also wondering if we should add extra info about cycles owned by the canister. according to the IC spec those are discarded. is that correct? 🤔 

see https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-delete_canister

Author: The cycles are returned to the controller's cycles balance, so they are not "discarded" in the sense they are burned. I think the spec wording for this is a bit misleading.
